src/main/java/ru/ayaz/UserHandler.java

```diff
@@ -2,130 +2,42 @@
 
 import java.io.BufferedReader;
 import java.io.IOException;
-import java.io.InputStreamReader;
-import java.io.PrintWriter;
-import java.net.Socket;
 
 
 public class UserHandler implements Runnable {
```

Comment:
Well, I see your point, you are right.
In this particular case with nickname you should use a setter (cause you want to create a User object and then ask his name), but all things being equal - prefer constructor.